.github/workflows/verifyGithubActionBuilds.yml

```diff
@@ -0,0 +1,26 @@
+name: Verify Github Action Builds
```

Comment:
Any reason for this being a new workflow and not part of the existing test workflow?

Reply:
Yeah, it's because it needs to checkout the head of the PR's branch, rather than the ref of the PR merged into master. I'm a little fuzzy on the git details of this, but when I had it in the existing test workflow it was showing a diff and failing because it was using the version # from package.json on master, even though it also had the latest changes from the PR.

So I think by default actions/checkout@v2 somehow pretends to merge the PR into master and checks out that ref, which is what we want for automated tests but not what we want for this.

Reply:
Well should we just be checking for a diff on the ncc built files? Not all files?

Reply:
That doesn't work because for some reason the npm version number ends up in `.github/actions/bumpVersion/index.js` after running `ncc`. Not sure why.

Reply:
Funny enough.. I think the script is working and it's highlighting an issue with the current source code.

It looks like since [this](https://github.com/Expensify/Expensify.cash/pull/1226/files#diff-b6bd93e0d81ffd41b79283c0268e77f7e75c18a2d34b84ef41dfd525e693907eR29) line is a `require`, `ncc` will "compile" that version for us.. 

For example:

```js
const currentPatchVersion = `v${__nccwpck_require__(306)/* .version.slice */ .i8.slice(0, -4)}`
```

```js

```/***/ 306:
/***/ ((module) => {

"use strict";
module.exports = {"i8":"1.0.1-355"};

/***/ }),

Which I am worried might break on future versions because it won't be recompiled for each new version...

Reply:
It looks like updating to using `fs` will not compile the version directly, for example

```js
const { version } = JSON.parse(fs.readFileSync('./package.json'))
```

Reply:
Okay, I see what you mean. Great detective work. Not sure if it would have really been a problem, but I think it's an improvement nonetheless.

Reply:
Can we try running this in `test.yml` now and see if it works?

Reply:
I updated the source code to use `fs` instead of `require`. I think it's okay to move this back to the regular `test` workflow. However, that workflow is titled "Jest Unit Tests". Do you think we should call it something else instead? We can also give each step it's own name - like "Verify Github Actions" and "Jest Unit Tests"

Reply:
We can do that 👍 - I am almost thinking we might want to combine E2E in `test.yml` eventually too. I am not sure we previously had a strategy for how to organize workflows, but I see the list growing and don't want to overload anything.

Reply:
Another strategy would be to have `test.yml` be the workflow titled "Automated Tests" or something, and then let that workflow have 3+ jobs:

1) Verify Github Actions
2) Jest Unit Tests
3) iOS E2E tests
4) Other E2E tests too maybe

From the Github Actions docs:

> By default, a workflow with multiple jobs will run those jobs in parallel.

Reply:
I was able to successfully move the "Verify Github Actions" workflow into "Automated Tests" workflow as a job like so:

![image](https://user-images.githubusercontent.com/47436092/105412083-14d5de00-5be9-11eb-9c59-c4d28bb046ef.png)

However, I ended up reverting this change because individual jobs cannot be restarted - whole workflows must be restarted. Therefore it's better to keep separate tests in separate workflows so they can be restarted separately. This will be more important for deploys and flaky E2E tests than this workflow, but it's a useful precedent nonetheless.